### ISSUE TITLE
[Feature request] View Angle Tool

### ISSUE THREAD

**Issue by febeslmeisl** [osmandapp/Osmand#8852]
I have written a small "View Angle Tool" on my local OsmAnd-Branch that I find very helpful for me and wanted to check, if this might also be of interest for the master-branch. The tool helps to identify far distance objects (like castles, towers, mountains, ...) on the map you see in real world.

Here a real life example scenario as a typical use case (please also refer to attached screenshots below) :

- I was on a bike tour, seeing a distant tower on a hill with radio antennas somewehere in the south and wanted to know what exactly it is.
- So I startet the tool, marked my exact position on the map (was standing on street crossing, see blue circle on the top left in [Step1.png](https://user-images.githubusercontent.com/34205180/80487368-fb0ab180-895c-11ea-9444-343a6641f3e4.png))
- From my point of view I saw the Tower right above a house (No 10 at the bottom) across a meadow, I could easily identify on the map
- So I marked the left and right limit in between I saw the tower from my point of view (see green and orange circle in  [Step1.png](https://user-images.githubusercontent.com/34205180/80487368-fb0ab180-895c-11ea-9444-343a6641f3e4.png))
(The left and right maker makes it easy the estimate the possible projection error, dependent on an inaccurate view position. Just step one or two steps to left and right check that tower is still within the marked limits in real world)
- Now I can zoom out ([Step2.png](https://user-images.githubusercontent.com/34205180/80487381-01009280-895d-11ea-8873-4be7fa982e9c.png)) (and even use OsmAnd search function to display e.g. all towers on the map).
- With the angle highlighted on the map and distance display generated by the tool it is easy to indentify a candidate on map (Tower near "Glems", see [Step2.png](https://user-images.githubusercontent.com/34205180/80487381-01009280-895d-11ea-8873-4be7fa982e9c.png) seems to be a good candidate)
- Zooming in ([Step3.png](https://user-images.githubusercontent.com/34205180/80487419-0e1d8180-895d-11ea-9d18-8f31f02d0c21.png)) shows a communication tower on 778m high hill ("Grasberg") in around 12.7km distance
- In this case I put it in my navigation as new target and rode my bike to that tower and had a beautiful view (and it actually was the tower I saw from the street crossing:))

Code would be ready for German and English language and also renderingStyles for default map (day and night) and topo. If you are interessted, please let me know if I should open an branch/PR for that and in case if you have any architecture, design or process guidelines to be followed for contributing to OsmAnd. 
It depends on Issue [#8835](https://github.com/osmandapp/Osmand/issues/8835).

Step1
![Step1](https://user-images.githubusercontent.com/34205180/80487368-fb0ab180-895c-11ea-9444-343a6641f3e4.png)
Step2
![Step2](https://user-images.githubusercontent.com/34205180/80487381-01009280-895d-11ea-8873-4be7fa982e9c.png)
Step3
![Step3](https://user-images.githubusercontent.com/34205180/80487419-0e1d8180-895d-11ea-9d18-8f31f02d0c21.png)

**Comment by vshcherb** [comment]
Tool looks really cool though I'm not sure if could be merged in the main branch especially because we plan to change that screen to fullfill other issues and design is not ready yet.

I would say it looks quite niche to the problem it solves, so I would be happy to see it as a plugin of course, though I'm not sure yet if we have enough api to support. As I see it consists mainly of 3 buttons

**Comment by Thunichgud** [comment]
This issue could be combined with the issue #7282: "[Feature Request] Bearing Mode ", which is assigned to be realised in Osmand 3.8.

**Comment by febeslmeisl** [comment]
> 
> 
> This issue could be combined with the issue #7282: "[Feature Request] Bearing Mode ", which is assigned to be realised in Osmand 3.8.

Yes, I agree. Both seem basically to cover the same use case. If we add the option to set bearing manually in #7282 - not only from (noisy) compass, but with selecting a precise point on map - it more or less covers my use case. Only (optional) things missing would be:
- View angle range (set two points as limit to estimate the error) -> bearing line could be generalized to be an angle around bearing line.
- A display of the distance while searching objects -> this could go into point 5 of #7282. Displaying besides bearing also the distance (from original point to map center while scrolling over the map to find objects along the bearing line/angle). Or as workaround use measurement tool, if this can be used in bearing mode.

**Comment by febeslmeisl** [comment]
If code is required for merging with #7282, it is now available on my Osmand fork:
- Code: [Branch Feature_8852_ViewAngleTool of Osmand](https://github.com/febeslmeisl/Osmand/tree/Feature_8852_ViewAngleTool)

- Rendering styles: [Branch Feature_8852_ViewAngleTool of OsmAnd-resources](https://github.com/febeslmeisl/OsmAnd-resources/tree/Feature_8852_ViewAngleTool)
